inferno-core/src/Inferno/Module/Prelude/Defs.hs

```diff
@@ -503,3 +507,33 @@ textSplitAt =
            in pure $ VTuple [VText a, VText b]
         _ -> throwError $ RuntimeError "splitAt: expecting text for the second argument"
     _ -> throwError $ RuntimeError "splitAt: expecting an int for the first argument"
+
+-- | Convert unsigned integer to binary-coded decimal.
+toBCD :: Word -> Word
+toBCD = digitSum 0 0
+  where
+    digitSum :: Word -> Int -> Word -> Word
+    digitSum acc _ x | x == 0 = acc
+    digitSum acc offset x =
+      let (q, r) = divMod x 10
+          acc' = unsafeShiftL r offset .|. acc
+       in acc' `seq` digitSum acc' (offset + 4) q
+
+toBCDFun :: Word64 -> Word64
+toBCDFun = fromIntegral . toBCD . fromIntegral
+
+-- | Convert binary-coded decimal to unsigned integer.
+fromBCD :: Word -> Maybe Word
+fromBCD = digitSeparation 0 1
+  where
+    digitSeparation :: Word -> Word -> Word -> Maybe Word
+    digitSeparation acc _ x | x == 0 = Just acc
+    digitSeparation acc offset x =
+      let d = x .&. 0b1111
+          acc' = acc + d * offset
+       in if d <= 9
+            then acc' `seq` digitSeparation acc' (offset * 10) (unsafeShiftR x 4)
+            else Nothing
+
+fromBCDFun :: Word64 -> Maybe Word64
+fromBCDFun = fmap fromIntegral . fromBCD . fromIntegral
```

Comment:
If you wanted to test these implementations, you could add some tests to `inferno-core/test/Eval/Spec.hs`

Reply:
@siddharth-krishna They already have their own tests in plow-math: https://github.com/plow-technologies/all/blob/master/plow-math/test/Main.hs#L55

That's where this code comes from.

Reply:
Ah I see, cool! I suppose ideally we'd use the plow-math implementation here too, but then we'd have to open source plow-math too, would that make sense? (cc @smurphy8 )

Reply:
I wouldn't open source the whole of plow-math, but I don't mind splitting the BCD stuff out and making that public.

Reply:
We could make a dev side issue to do that but I don’t want to mess with it right now.